master/pkg/schemas/expconf/environment_config.go

```diff
@@ -21,17 +22,30 @@ func (p PodSpec) Copy() PodSpec {
 	return PodSpec(*k8sP.DeepCopy())
 }
 
-// Merge implements the schemas.Mergable psuedointerface.
+// Merge implements the schemas.Mergable psuedointerface using Kubernetes strategic merging.
+//nolint:lll
+// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/strategic-merge-patch.md
```

Comment:
This says strategic merging is obsolete; is it truly what we want? Does it end up being equivalent to the replacement "Server-Side Apply" or something?

Reply:
I agree that it is somewhat concerning it says "probably obsolete". It is referenced without any warning though in other docs here https://kubernetes.io/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/#use-a-strategic-merge-patch-to-update-a-deployment

Though I'm not sure if "Server-Side Apply" is ideal either, since I believe it would require us submitting the pod spec to the api server then calling a patch adding some complexity, especially around what fields we allow to be patched. I don't think the benefits of "Server-Side Apply" really help us too https://kubernetes.io/blog/2022/10/20/advanced-server-side-apply/#benefits (conflicts detection don't apply).


Any other options you can think of? Since taking a step back all we want from this feature is just to reasonably merge pod specs. I don't think that json merge patch would produce ideal results since it would require specifying whole lists to replace.